crimethinc/website
# started working on dockerization

# How does this pull request make you feel (in animated GIF format)?

![do it live](https://media.giphy.com/media/q7UpJegIZjsk0/giphy.gif)

# What are the relevant GitHub issues?

resolves [Issue 1823](https://github.com/crimethinc/website/issues/1823)

# What does this pull request do?

This PR introduces three new files and some minor modifications which I believed necessary to a script or two...
1) `Dockerfile` which contains the image definition for the rails service itself
2) `docker-compose.yml` which contains the compose yaml for launching the rails service *and* the postgres instance separately.
3) `config/environments/docker.rb` which is basically a clone of `development.rb` but `database.yml` specifies the username for the database to be `postgres`. This was the most straight forward way to do this without having to do complicated changes on dockers side.

Another thing to note, if you see `docker-compose.yml` you'll see an environment variable specified named `POSTGRES_HOST_AUTH_METHOD` which is set to `trust`. This just makes it so we don't need to auth if we're connecting locally, from what I understand, but I truly don't fully understand this flag in its entirety so we probably shouldn't deploy this until we make sure we know what we're doing here. :P

I've added instructions on how to use this in the README but basically you just need to run the following two commands and it *should* work, considering any previous setup necessary depending on your distro/os.

```sh
docker build -t crimethinc .
docker compose up
```

p.s

The database and rails service being separate is necessary. Docker containers are designed to run one thing at a time. It's not impossible to run the database and rails service together but it is not a good idea. :D

# How should this be manually tested?

Depends on how we deploy but basically, everything should be tested as normal, manually. Probably scrape the site and make sure everything is accessible just the same. Should be straight forward. We can probably test it however you have tested it before.

# Is there any background context you want to provide for reviewers?
See above. There's a bit to consider here but I got it working to its current state and looking forward to discussing what we could do to make it better.

# Questions for the pull request author (delete any that don't apply):
- [ ] Does this pull request require any new server dependencies which need to be added to the build process? If so, please explain what.

# Acceptance Criteria
## These should be checked by the reviewers
